nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -129,26 +144,44 @@
             .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
             .build();
     static final PropertyDescriptor RESULTS_PER_FLOWFILE = new PropertyDescriptor.Builder()
-            .name("results-per-flowfile")
-            .displayName("Results Per FlowFile")
-            .description("How many results to put into a flowfile at once. The whole body will be treated as a JSON array of results.")
-            .required(false)
-            .expressionLanguageSupported(true)
-            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
-            .build();
+        .name("results-per-flowfile")
+        .displayName("Results Per FlowFile")
+        .description("How many results to put into a flowfile at once. The whole body will be treated as a JSON array of results.")
+        .required(false)
+        .expressionLanguageSupported(true)
+        .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+        .build();
+    static final PropertyDescriptor ESTIMATE_PROGRESS = new PropertyDescriptor.Builder()
+        .name("estimate-progress")
+        .displayName("Estimate Progress")
+        .description("If enabled, a count query will be run first, using the configured query, and attributes will be added to each flowfile showing how far they are into the result set.")
+        .required(true)
+        .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+        .allowableValues(GM_TRUE, GM_FALSE)
+        .defaultValue(GM_FALSE.getValue())
+        .build();
+    static final PropertyDescriptor PROGRESSIVE_COMMITS = new PropertyDescriptor.Builder()
+        .name("progressive-commits")
+        .displayName("Commit After Each Batch")
```

Comment:
I'm a little confused here about the term "batch". It doesn't seem directly related to the Batch Size property (since the latter is kind of a server-side thing, like a JDBC "fetch size"?), and in the code a "batch" seems to refer to the number of files set in Results Per Flowfile.
Can you explain a little more about what's going on with the progressive commits? If I have Results per Flowfile set to 100 and Batch Size set to 1000, would I get 10 flow files committed at once as once "batch"? Or is it always one commit per flowfile (if Commit After Each Batch is set)?

Reply:
It works in coordination with the results per flowfile property. The idea is to emulate the ExecuteSQL where after a batch of X number has been built up in the processor and it sends the data to a flowfile, it commits.

I'm tempted to change the batch size property's display name to be something like Query Fetch Size. I think Results Per Flowfile is probably even clearer than Batch Size for this. Thoughts?

Reply:
I spent a little time to do working over the documentation on the properties, and I think I struck the right balance of renaming/redocumenting. As an added benefit, I think I made the Batch Size property (now Fetch Size) much clearer to newbs.